### PR TITLE
docs: document Deno's HTTP Server API

### PR DESCRIPTION
**Pull request by kitsonk**
Convert the example HTTP Server to use the Deno HTTP Server API and add a section to the Deno APIs about the HTTP Server APIs.

### CODE REVIEW

**Review by lucacasonato — APPROVED** [review]
Looks great. Documentation is clear and highly linkable, which is great.

**Comment by kitsonk** [comment]
I need to clear my local dprint and relint to fix the lint issue.

**Review by ry — APPROVED** [review]
Nice - thanks!

File: docs/examples/http_server.md [hunk 1]
```diff
@@ -2,25 +2,75 @@
...
 
-```typescript
-/**
- * webserver.ts
- */
+**webserver.ts**:
+
+```ts
+// Start listening on port 8080 of localhost.
+const server = Deno.listen({ port: 8081 });
```
Comment (satyarohith) [inline docs/examples/http_server.md:25]:
```suggestion
const server = Deno.listen({ port: 8080 });
```

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**